lib/OpenQA/Utils.pm

```diff
@@ -112,9 +119,12 @@ sub needle_info($$$) {
         return;
     }
 
+    my $fname = basename($jsonfile, '.json') . ".png";
+    my $pngfile = File::Spec->catpath('', dirname($jsonfile), $fname);
```

Comment:
This block is confusing to me:
1) why are you using two identical variables `$jsonfile` and `$fn`?
2) in eval exception handler you don't use any of them and instead again cat `$needledir/$name.json`
3) use `dirname($jsonfile)` instead of `$needledir`

When you'll be touching the eval exception handler, I would appreciate to use `Try::Tiny` and change `eval {..} if {..}` to `try {..} catch {..};`